routing/get_fees_paid.js

```diff
@@ -90,6 +93,56 @@ module.exports = (args, cbk) => {
       // Get node icons
       getIcons: ['validate', ({}, cbk) => getIcons({fs: args.fs}, cbk)],
 
+      // Determine the In public key to use
+      getInKey: ['validate', (({}, cbk) => {
+        // Exit early if no in query is specified
+        if (!args.in) {
+          return cbk();
+        }
+
+        return asyncMap(args.lnds, (lnd, cbk) => {
+          return findKey({lnd, query: args.in}, cbk);
+        },
+        (err, res) => {
+          if (!!err) {
+            return cbk(err);
+          }
+
+          if (res.length > 1) {
+            return cbk([400, 'MultipleMatchesForInQuery']);
+          }
+
+          const key = res[0].public_key;
+
+          return cbk(null, {key});
+        });
+      })],
+
+      // Determine the Out public key to use
+      getOutKey: ['validate', (({}, cbk) => {
+        // Exit early if no out query is specified
+        if (!args.out) {
+          return cbk();
+        }
+
+        return asyncMap(args.lnds, (lnd, cbk) => {
+          return findKey({lnd, query: args.out}, cbk);
+        },
+        (err, res) => {
+          if (!!err) {
+            return cbk(err);
+          }
+
+          if (res.length > 1) {
+            return cbk([400, 'MultipleMatchesForOutQuery']);
```

Comment:
Find key may return an empty object, like {}, so need to make sure it errors if there is no match on either